src/mr.rs

```diff
@@ -0,0 +1,499 @@
+use crate::{
+    device::{
+        ToCardCtrlRbDesc, ToCardCtrlRbDescCommon, ToCardCtrlRbDescUpdateMrTable,
+        ToCardCtrlRbDescUpdatePageTable,
+    },
+    responser::AcknowledgeBuffer,
+    types::{Key, MemAccessTypeFlag, PAGE_SIZE},
+    utils::{allocate_aligned_memory, deallocate_aligned_memory},
+    Device, Error, Pd,
+};
+use rand::RngCore as _;
+use std::{
+    hash::{Hash, Hasher},
+    mem, ptr,
+    sync::Arc,
+};
+
+const ACKNOWLEDGE_BUFFER_SLOT_CNT: usize = 1024;
+const ACKNOWLEDGE_BUFFER_SIZE: usize =
+    ACKNOWLEDGE_BUFFER_SLOT_CNT * AcknowledgeBuffer::ACKNOWLEDGE_BUFFER_SLOT_SIZE;
+
+/// Memory Region
+///
+/// User use `Device::alloc_mr(..)` to allocate a `Mr` and use `Device::dereg_mr(..)` to deallocate a `Mr`.
+#[derive(Debug, Clone, Copy)]
+pub struct Mr {
+    pub(crate) key: Key,
+}
+
+impl Mr {
+    /// Get the key of the Mr
+    #[must_use]
+    pub fn get_key(&self) -> Key {
+        self.key
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct MrCtx {
+    #[allow(unused)]
```

Comment:
Please comment the reason of this unused code, so that we can tell when to remove it later.

Reply:
Sure, I will do that.